### PR TITLE
Unified printing

### PR DESCRIPTION
**Pull request by leios**
This is a functioning version for unified printing for CUDA and CPU

### CODE REVIEW

File: src/KernelAbstractions.jl [hunk 1]
```diff
@@ -345,6 +388,10 @@ function __synchronize()
     error("@synchronize used outside kernel or not captured")
 end
 
+function __print(items...)
+    error("@print used outside of kernel or not captured")
```
Comment (vchuravy) [inline src/KernelAbstractions.jl]:
```suggestion
    error("@print used outside of kernel")
```

File: src/macros.jl [hunk 2]
```diff
@@ -92,9 +92,9 @@ function transform_cpu!(def, constargs)
     end
     new_stmts = Expr[]
     body = MacroTools.flatten(def[:body])
-    push!(new_stmts, Expr(:aliasscope))
+    #push!(new_stmts, Expr(:aliasscope))
```
Comment (vchuravy) [inline src/macros.jl]:
```suggestion
    push!(new_stmts, Expr(:aliasscope))
```

File: src/macros.jl [hunk 3]
```diff
@@ -92,9 +92,9 @@ function transform_cpu!(def, constargs)
...
     append!(new_stmts, split(body.args))
-    push!(new_stmts, Expr(:popaliasscope))
+    #push!(new_stmts, Expr(:popaliasscope))
```
Comment (vchuravy) [inline src/macros.jl]:
```suggestion
    push!(new_stmts, Expr(:popaliasscope))
```

File: test/print_test.jl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    using CuArrays
+    CuArrays.allowscalar(false)
+end
+
+@kernel function kernel_print()
+  I = @index(Global)
+  @print("Hello from thread ", I, "!\n")
+end
+
+function test_print(gpu_avail)
```
Comment (vchuravy) [inline test/print_test.jl]: Maybe pass in the backend here, instead of using a boolean to switch?

**Comment by leios** [comment]
bors try

**Pull request merged**